samza-core/src/test/java/org/apache/samza/operators/TestStreamGraphSpec.java

```diff
@@ -520,4 +434,42 @@ public void testGetTable() {
         new TableSpec("t1", KVSerde.of(new NoOpSerde(), new NoOpSerde()), "", new HashMap<>()));
     assertNotNull(graphSpec.getTable(mockTableDescriptor));
   }
+
```

Comment:
I might have missed it, but did you add a test for when the same system name is used for multiple descriptors?